File: KSTests/InitTest.vb

'    Kernel Simulator  Copyright (C) 2018-2022  Aptivi
'
'    This file is part of Kernel Simulator
'
'    Kernel Simulator is free software: you can redistribute it and/or modify
'    it under the terms of the GNU General Public License as published by
'    the Free Software Foundation, either version 3 of the License, or
'    (at your option) any later version.
'
'    Kernel Simulator is distributed in the hope that it will be useful,
'    but WITHOUT ANY WARRANTY; without even the implied warranty of
'    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
'    GNU General Public License for more details.
'
'    You should have received a copy of the GNU General Public License
'    along with this program.  If not, see <https://www.gnu.org/licenses/>.

Imports System.IO
Imports KS.Files.Querying
Imports KS.Misc.Configuration

<SetUpFixture> Public Class InitTest

    ''' <summary>
    ''' Initialize everything that is required before starting unit tests
    ''' </summary>
    <OneTimeSetUp> Public Shared Sub ReadyEverything()
        InitPaths()
        If Not FileExists(GetKernelPath(KernelPathType.Configuration)) Then
            CreateConfig()
        Else
            If Not FileExists(GetKernelPath(KernelPathType.Configuration) + ".old") Then File.Move(GetKernelPath(KernelPathType.Configuration), GetKernelPath(KernelPathType.Configuration) + ".old")
            CreateConfig()
        End If
        InitializeConfigToken()
        LoadUserToken()

        'NUnit sets current directory to a wrong directory, so set it to the test context directory
        Dim TestAssemblyDir As String = TestContext.CurrentContext.TestDirectory
        Environment.CurrentDirectory = TestAssemblyDir
    End Sub

    ''' <summary>
    ''' Clean up everything that the unit tests made
    ''' </summary>
    <OneTimeTearDown> Public Shared Sub CleanEverything()
        If FileExists(HomePath + "/Documents/TestText.txt") Then File.Delete(HomePath + "/Documents/TestText.txt")
        If FileExists(HomePath + "/Documents/Text.txt") Then File.Delete(HomePath + "/Documents/Text.txt")
        If FileExists(HomePath + "/NewFile.txt") Then File.Delete(HomePath + "/NewFile.txt")
        If FileExists(HomePath + "/NewFile.json") Then File.Delete(HomePath + "/NewFile.json")
        If FileExists(HomePath + "/1mb-test.csv") Then File.Delete(HomePath + "/1mb-test.csv")
        If FolderExists(HomePath + "/TestMovedDir2") Then Directory.Delete(HomePath + "/TestMovedDir2", True)
        If FolderExists(HomePath + "/NewDirectory") Then Directory.Delete(HomePath + "/NewDirectory", True)
        If FolderExists(HomePath + "/TestDir") Then Directory.Delete(HomePath + "/TestDir", True)
        If FolderExists(HomePath + "/TestDir2") Then Directory.Delete(HomePath + "/TestDir2", True)
        If FileExists(GetKernelPath(KernelPathType.Configuration) + ".old") Then
            If FileExists(GetKernelPath(KernelPathType.Configuration)) Then File.Delete(HomePath + "/KernelConfig.json")
            File.Move(GetKernelPath(KernelPathType.Configuration) + ".old", GetKernelPath(KernelPathType.Configuration))
        End If
    End Sub

End Class

File: private/KSTests/InitTest.cs

// Nitrocid KS  Copyright (C) 2018-2023  Aptivi
// 
// This file is part of Nitrocid KS
// 
// Nitrocid KS is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
// 
// Nitrocid KS is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
// 
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.

using System;
using System.IO;
using KS.Files;
using KS.Files.Querying;
using KS.Kernel.Configuration;
using NUnit.Framework;
using KS.Users;
using KS.Files.Operations;

namespace KSTests
{
    [SetUpFixture]
    public class InitTest
    {
        /// <summary>
        /// Initialize everything that is required before starting unit tests
        /// </summary>
        [OneTimeSetUp]
        public static void ReadyEverything()
        {
            if (!Checking.FileExists(Paths.GetKernelPath(KernelPathType.Configuration)))
            {
                // Check to see if we have an appdata folder for KS
                if (!Checking.FolderExists(Paths.AppDataPath))
                    Making.MakeDirectory(Paths.AppDataPath, false);

                // Now, create config
                Config.CreateConfig();
            }
            else
            {
                if (!Checking.FileExists(Paths.GetKernelPath(KernelPathType.Configuration) + ".old"))
                    File.Move(Paths.GetKernelPath(KernelPathType.Configuration), Paths.GetKernelPath(KernelPathType.Configuration) + ".old");
                Config.CreateConfig();
            }
            Config.ReadConfig(ConfigType.Kernel, Paths.GetKernelPath(KernelPathType.Configuration));

            // NUnit sets current directory to a wrong directory, so set it to the test context directory
            string TestAssemblyDir = TestContext.CurrentContext.TestDirectory;
            Environment.CurrentDirectory = TestAssemblyDir;
        }

        /// <summary>
        /// Clean up everything that the unit tests made
        /// </summary>
        [OneTimeTearDown]
        public static void CleanEverything()
        {
            if (Checking.FileExists(Paths.HomePath + "/Documents/TestText.txt"))
                File.Delete(Paths.HomePath + "/Documents/TestText.txt");
            if (Checking.FileExists(Paths.HomePath + "/Documents/Text.txt"))
                File.Delete(Paths.HomePath + "/Documents/Text.txt");
            if (Checking.FileExists(Paths.HomePath + "/NewFile.txt"))
                File.Delete(Paths.HomePath + "/NewFile.txt");
            if (Checking.FileExists(Paths.HomePath + "/NewFile.json"))
                File.Delete(Paths.HomePath + "/NewFile.json");
            if (Checking.FileExists(Paths.HomePath + "/1mb-test.csv"))
                File.Delete(Paths.HomePath + "/1mb-test.csv");
            if (Checking.FolderExists(Paths.HomePath + "/TestMovedDir2"))
                Directory.Delete(Paths.HomePath + "/TestMovedDir2", true);
            if (Checking.FolderExists(Paths.HomePath + "/NewDirectory"))
                Directory.Delete(Paths.HomePath + "/NewDirectory", true);
            if (Checking.FolderExists(Paths.HomePath + "/TestDir"))
                Directory.Delete(Paths.HomePath + "/TestDir", true);
            if (Checking.FolderExists(Paths.HomePath + "/TestDir2"))
                Directory.Delete(Paths.HomePath + "/TestDir2", true);
            if (Checking.FileExists(Paths.GetKernelPath(KernelPathType.Configuration) + ".old"))
            {
                if (Checking.FileExists(Paths.GetKernelPath(KernelPathType.Configuration)))
                    File.Delete(Paths.GetKernelPath(KernelPathType.Configuration));
                File.Move(Paths.GetKernelPath(KernelPathType.Configuration) + ".old", Paths.GetKernelPath(KernelPathType.Configuration));
            }
        }
    }
}
